delta/plugins/search/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/SearchConfigUpdater.scala

```diff
@@ -0,0 +1,112 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.search
+
+import cats.Eq
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.CompositeViews
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef.ActiveViewDef
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewValue.indexingEq
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.{CompositeViewFields, CompositeViewValue}
+import ch.epfl.bluebrain.nexus.delta.plugins.search.SearchScopeInitialization._
+import ch.epfl.bluebrain.nexus.delta.plugins.search.model.SearchConfig.IndexingConfig
+import ch.epfl.bluebrain.nexus.delta.plugins.search.model.{defaultViewId, SearchConfig}
+import ch.epfl.bluebrain.nexus.delta.sdk.Defaults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.BaseUri
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ElemStream
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream._
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.Task
+
+/**
+  * Allows to update the search config of default composite views. The provided defaults and indexing config provide the
+  * basis of the [[CompositeViewFields]] to which the composite views are compared.
+  */
+final class SearchConfigUpdater(defaults: Defaults, config: IndexingConfig) {
+
+  /**
+    * For the given composite views, updates the active ones if their search config differs from the current one.
+    * @param views
+    *   a stream of views to perform the update on
+    * @param update
+    *   a function that defines what update should be done to an active view
+    */
+  def apply(
+      views: ElemStream[CompositeViewDef],
+      update: (ActiveViewDef, CompositeViewFields) => Task[Unit]
+  ): Stream[Task, Elem[CompositeViewDef]] =
+    views
+      .filter(_.id == defaultViewId)
+      .evalTap { elem =>
+        elem.traverse {
+          case view: ActiveViewDef if configHasChanged(view) =>
+            update(view, defaultSearchViewFields)
+          case _                                             =>
+            Task.unit
+        }
+      }
+
+  private def configHasChanged(
+      v: ActiveViewDef
+  )(implicit eq: Eq[CompositeViewValue] = indexingEq): Boolean =
+    v.value =!= defaultSearchViewValue(v)
```

Comment:
If the equality is never different then it's probably better to remove it from the method signature, just to keep the surface area smaller